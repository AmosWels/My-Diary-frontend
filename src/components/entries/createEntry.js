import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CreateView from "../../containers/entries/createEntryContainer";
import { createEntryAction } from "../../actions/diaries/diariesAction";
class CreateEntryModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false,
      due_date: "",
      name: "",
      purpose: "",
      type: ""
    };
  }

  toggle = () => {
    this.setState({
      modal: !this.state.modal
    });
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = () => {
    const data = {
      due_date: this.state.due_date,
      name: this.state.name,
      purpose: this.state.purpose,
      type: this.state.type
    };
    this.props.dispatch(createEntryAction(data));
  };
  modalClose = () => {
    this.toggle();
    this.handleSubmit();
  };

  render() {
    const { modal, due_date, name, purpose, type } = this.state;
    const classname = this.props.className;
    return (
      <div>
        <CreateView
          toggle={this.toggle}
          modal={modal}
          name={name}
          due_date={due_date}
          purpose={purpose}
          type={type}
          classname={classname}
          closemodal={this.modalClose}
          handleChange={this.handleChange}
        />
      </div>
    );
  }
}
CreateEntryModal.propTypes = {
  emailShare: PropTypes.func,
  className: PropTypes.string,
  dispatch: PropTypes.func
};
const mapDispatchToProps = dispatch => ({ dispatch });
const mapStateToProps = state => ({
  articles: state.articlesReducer
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateEntryModal);
