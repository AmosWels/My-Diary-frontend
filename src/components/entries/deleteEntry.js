import React, { Component } from "react";
import PropTypes from "prop-types";
import { confirmAlert } from "react-confirm-alert"; // Import
import "react-confirm-alert/src/react-confirm-alert.css"; // Import css
import { CardLink } from "reactstrap";

class DeleteEntryConfirm extends Component {
  submit = (id) => {
    confirmAlert({
      title: "Confirm to Delete",
      message: "Are you sure to do this.",
      buttons: [
        {
          label: "Yes",
          onClick: () => this.props.handleDelete(this.props.uniqueId)
        },
        {
          label: "No"
        }
      ]
    });
  };

  render() {
    return <CardLink id="cardlink" onClick={this.submit} href="#">Delete |</CardLink>;
  }
}
DeleteEntryConfirm.propTypes = {
  handleDelete: PropTypes.func,
  uniqueId: PropTypes.id
};
export default DeleteEntryConfirm;
