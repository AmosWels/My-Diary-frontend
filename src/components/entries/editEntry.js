import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import EditView from "../../containers/entries/editEntryContainer";
import { updateEntryAction } from "../../actions/diaries/diariesAction";

class EditEntryModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false,
      due_date: this.props.due_date,
      name: this.props.name,
      purpose: this.props.purpose,
      type: this.props.type
    };
  }

  toggle = () => {
    this.setState({
      modal: !this.state.modal
    });
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleUpdate = () => {
    const data = {
      due_date: this.state.due_date,
      name: this.state.name,
      purpose: this.state.purpose,
      type: this.state.type
    };
    this.props.dispatch(updateEntryAction(this.props.uniqueId,data));
  };
  modalClose = () => {
    this.toggle();
    this.handleUpdate();
  };

  render() {
    const { modal, due_date, name, purpose, type } = this.state;
    const classname = this.props.className;
    return (
      <React.Fragment>
        <EditView
          toggle={this.toggle}
          modal={modal}
          name={name}
          due_date={due_date}
          purpose={purpose}
          type={type}
          classname={classname}
          closemodal={this.modalClose}
          handleChange={this.handleChange}
          props={this.props}
        />
      </React.Fragment>
    );
  }
}
EditEntryModal.propTypes = {
  emailShare: PropTypes.func,
  className: PropTypes.string,
  dispatch: PropTypes.func,
  props: PropTypes.object,
  uniqueId:PropTypes.string
};
const mapDispatchToProps = dispatch => ({ dispatch });
const mapStateToProps = state => ({
  articles: state.articlesReducer
});
export {EditEntryModal as EditEntryModalTest};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditEntryModal);
