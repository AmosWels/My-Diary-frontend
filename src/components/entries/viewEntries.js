import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { getAllEntries } from "../../actions/diaries/diariesAction";
import EntriesView from "../../containers/entries/entriesContainer";

class ViewDiaries extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.props.dispatch(getAllEntries());
  }

  render() {
    const entries = this.props.allentries;
    return (
      <div>
        <EntriesView results={entries} />
      </div>
    );
  }
}
ViewDiaries.propTypes = {
  dispatch: PropTypes.func.isRequired,
  allentries: PropTypes.object
};
const mapStateToProps = state => ({
  allentries: state.entriesReducer.entries
});
const mapDispatchToProps = dispatch => ({ dispatch });
export { ViewDiaries as ViewDiariesTest };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ViewDiaries);
