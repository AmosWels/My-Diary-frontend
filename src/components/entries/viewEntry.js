import React from "react";
import {
  CardLink,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter,
  Badge
} from "reactstrap";
import PropTypes from "prop-types";

class ViewEntry extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    const closeBtn = (
      <button className="close" onClick={this.toggle}>
        &times;
      </button>
    );
    return (
      <React.Fragment>
        <CardLink id="cardlink" onClick={this.toggle} href="#">
          View |
        </CardLink>
        <Modal
          isOpen={this.state.modal}
          toggle={this.toggle}
          className={this.props.className}
        >
          <ModalHeader toggle={this.toggle} close={closeBtn}>
            {this.props.name}
          </ModalHeader>
          <ModalBody id="scrollview">{this.props.purpose}</ModalBody>
          <ModalFooter>
            <Badge color="warning" className="float-left">
              Due Date - {this.props.due_date}
            </Badge>{" "}
            <Badge color="info" className="float-left">
              Created On - {this.props.date_created}
            </Badge>
          </ModalFooter>
        </Modal>
      </React.Fragment>
    );
  }
}
ViewEntry.propTypes = {
  buttonLabel: PropTypes.string,
  className: PropTypes.string,
  name: PropTypes.string,
  purpose: PropTypes.string,
  due_date: PropTypes.string,
  date_created: PropTypes.string
};

export default ViewEntry;
