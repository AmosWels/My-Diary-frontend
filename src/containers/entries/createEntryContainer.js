import React from "react";
import PropTypes from "prop-types";
import {
  Button,
  Label,
  Form,
  FormGroup,
  Input,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter
} from "reactstrap";

const CreateView = ({
  toggle,
  modal,
  classname,
  closemodal,
  handleChange,
  name,
  due_date,
  purpose,
  type
}) => {
  return (
    <div>
      <a target="_blank" onClick={toggle}>
        <Button block color="link" id="btn_link">
          Create Entry
        </Button>
      </a>
      <Modal isOpen={modal} toggle={toggle} className={classname}>
        <ModalHeader toggle={toggle}>Create New Entry</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup>
              <Label for="exampleEmail">Name</Label>
              <Input
                type="email"
                name="name"
                id="exampleEmail"
                placeholder="enter name of entry ..."
                onChange={handleChange}
                value={name}
                required
              />
            </FormGroup>
            <FormGroup>
              <Label for="exampleDate">Due Date</Label>
              <Input
                type="text"
                name="due_date"
                id="exampleDate"
                placeholder="YYYY-MM-DD"
                onChange={handleChange}
                value={due_date}
                required
              />
            </FormGroup>
            <FormGroup>
              <Label for="exampleEmail">Type</Label>
              <Input
                type="text"
                name="type"
                id="exampleEmail"
                placeholder="enter type of entry ..."
                onChange={handleChange}
                value={type}
                required
              />
            </FormGroup>
            <FormGroup>
              <Label for="exampleText">Purpose</Label>
              <Input
                type="textarea"
                name="purpose"
                id="exampleText"
                onChange={handleChange}
                value={purpose}
                required
              />
            </FormGroup>
          </Form>
        </ModalBody>
        <ModalFooter>
          <Button className="pull-right" onClick={closemodal} type="submit">
            Create
          </Button>
        </ModalFooter>
      </Modal>
    </div>
  );
};
CreateView.propTypes = {
  closemodal: PropTypes.func,
  handleChange: PropTypes.func,
  toggle: PropTypes.func,
  modal: PropTypes.bool,
  classname: PropTypes.string,
  name: PropTypes.string,
  due_date: PropTypes.string,
  type: PropTypes.string,
  purpose: PropTypes.string
};
export default CreateView;
