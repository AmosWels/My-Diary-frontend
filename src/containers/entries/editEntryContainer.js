import React from "react";
import PropTypes from "prop-types";
import {
  Button,
  Label,
  Form,
  FormGroup,
  Input,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter,
  CardLink
} from "reactstrap";

const EditView = ({
  toggle,
  modal,
  classname,
  closemodal,
  handleChange,
  name,
  due_date,
  purpose,
  type
}) => {
  return (
    <React.Fragment>
      <CardLink id="cardlink" onClick={toggle} href="#">
        Edit |
      </CardLink>
      <Modal isOpen={modal} toggle={toggle} className={classname}>
        <ModalHeader toggle={toggle}>Edit Entry</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup>
              <Label for="exampleEmail">Name</Label>
              <Input
                type="email"
                name="name"
                id="exampleEmail"
                placeholder="enter name of entry ..."
                onChange={handleChange}
                value={name}
                required
              />
            </FormGroup>
            <FormGroup>
              <Label for="exampleDate">Due Date</Label>
              <Input
                type="text"
                name="due_date"
                id="exampleDate"
                placeholder="YYYY-MM-DD"
                onChange={handleChange}
                value={due_date}
                required
              />
            </FormGroup>
            <FormGroup>
              <Label for="exampleEmail">Type</Label>
              <Input
                type="text"
                name="type"
                id="exampleEmail"
                placeholder="enter type of entry ..."
                onChange={handleChange}
                value={type}
                required
              />
            </FormGroup>
            <FormGroup>
              <Label for="exampleText">Purpose</Label>
              <Input
                type="textarea"
                name="purpose"
                id="exampleText"
                onChange={handleChange}
                value={purpose}
                required
              />
            </FormGroup>
          </Form>
        </ModalBody>
        <ModalFooter>
          <Button className="pull-right" onClick={closemodal} type="submit">
            Update
          </Button>
        </ModalFooter>
      </Modal>
    </React.Fragment>
  );
};
EditView.propTypes = {
  closemodal: PropTypes.func,
  handleChange: PropTypes,
  toggle: PropTypes.func,
  modal: PropTypes.string,
  classname: PropTypes.string,
  name: PropTypes.string,
  due_date: PropTypes.string,
  type: PropTypes.string,
  purpose: PropTypes.string
};
export default EditView;
