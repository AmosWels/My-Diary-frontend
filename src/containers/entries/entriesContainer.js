import React from "react";
import PropTypes from "prop-types";
import NavBar from "../../navigation/navBar";
import DeleteEntryConfirm from "../../components/entries/deleteEntry";
import ViewEntry from "../../components/entries/viewEntry";
import {
  Badge,
  Card,
  CardLink,
  CardBody,
  CardHeader,
  Col,
  CardFooter
} from "reactstrap";

const EntriesView = ({ results, handleDelete }) => {
  let showEntries;
  if (!(results instanceof Array)) {
    showEntries = (
      <Col xs="12" sm="12" md="12" id="column_gap">
        <Card className="card-accent-primary border-primary">
          <CardHeader>Hello, Welcome</CardHeader>
          <CardBody>
            <center>
              <Badge color="danger" className="" id="entrymessage">
                You dont have any entries yet, please first create
              </Badge>
            </center>
          </CardBody>
        </Card>
      </Col>
    );
  } else {
    showEntries = results.map(entry => (
      <Col xs="12" sm="6" md="4" key={entry.id} id="column_gap">
        <Card className="card-accent-primary border-primary">
          <CardHeader>
            {entry.name}{" "}
            <div className="d-inline actions">
              <CardLink id="cardlink" href="#">
                Edit |
              </CardLink>
              <ViewEntry
                name={entry.name}
                purpose={entry.purpose}
                due_date={entry.due_date}
                type={entry.type}
                date_created={entry.date_created}
              />
              <DeleteEntryConfirm
                handleDelete={handleDelete}
                uniqueId={entry.id}
              />
            </div>
          </CardHeader>
          <Badge color="warning" className="float-left">
            Due Date - {entry.due_date}
          </Badge>
          <CardBody>{entry.purpose.substring(0, 62)} . . .</CardBody>
          <CardFooter>
            <Badge color="info" className="float-left">
              Created On - {entry.date_created}
            </Badge>
            <Badge color="info" className="float-right">
              Type - {entry.type}
            </Badge>
          </CardFooter>
        </Card>
      </Col>
    ));
  }
  return (
    <div>
      <div className="containers">
        <NavBar />
        <br />
        <div className="pic-container">{showEntries}</div>
      </div>
    </div>
  );
};
EntriesView.propTypes = {
  results: PropTypes.array,
  handleSignup: PropTypes.func,
  handleDelete: PropTypes.func
};

export default EntriesView;
