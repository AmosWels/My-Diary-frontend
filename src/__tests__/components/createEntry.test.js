import React from "react";
import { shallow, configure } from "enzyme";
import configureMockStore from "redux-mock-store";
import { SignupCreateEntryModalTest } from "../../components/entries/createEntry";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("SignupUser component", () => {
  const history = {push : jest.fn()};
  const props = {
    history,
    handleSubmit: jest.fn(),
    modalClose: jest.fn(),
    dispatch: jest.fn(),
    handleChange:jest.fn(),
    toggle:jest.fn(),
    name: "",
    purpose: "",
    due_date: "",
    type: "",
  };
  const getEvent = (name = "", value = "") => ({
    preventDefault: jest.fn(),
    target: {
      name,
      value
    }
  });
  const mockStore = configureMockStore();
  let wrapper;

  beforeEach(() => {
    const store = mockStore({});
    wrapper = shallow(<SignupCreateEntryModalTest {...props} store={store} />);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should set state when handle change is called", () => {
    wrapper.instance().handleChange(getEvent("purpose", "home"));
    expect(wrapper.state().username).toBeUndefined();
  });

  it("should handle submit when called", () => {
    wrapper.instance().handleSubmit(getEvent());
    expect(wrapper.state().username).toBeUndefined();
  });

  it("should not redirect after submit failure", () => {
    const nextprops = {
      user:{
        user:{ Message:"Created Succesfull"}
      }
    };
    wrapper.setProps({...nextprops});
    expect(history.push).not.toBeCalled();
  });
});
