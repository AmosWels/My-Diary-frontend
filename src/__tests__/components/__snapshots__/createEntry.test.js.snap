// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupUser component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateEntryModal
    dispatch={[MockFunction]}
    due_date=""
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    modalClose={[MockFunction]}
    name=""
    purpose=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    toggle={[MockFunction]}
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CreateView
        closemodal={[Function]}
        due_date=""
        handleChange={[Function]}
        modal={false}
        name=""
        purpose=""
        toggle={[Function]}
        type=""
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classname": undefined,
        "closemodal": [Function],
        "due_date": "",
        "handleChange": [Function],
        "modal": false,
        "name": "",
        "purpose": "",
        "toggle": [Function],
        "type": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CreateView
          closemodal={[Function]}
          due_date=""
          handleChange={[Function]}
          modal={false}
          name=""
          purpose=""
          toggle={[Function]}
          type=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "classname": undefined,
          "closemodal": [Function],
          "due_date": "",
          "handleChange": [Function],
          "modal": false,
          "name": "",
          "purpose": "",
          "toggle": [Function],
          "type": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
